tools/http-fault-injector/sample-clients/java/storage-blobs/src/main/java/App.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Configuration;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobClientBuilder;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import com.azure.storage.common.policy.RetryPolicyType;
+
+import java.time.Duration;
+
+public class App {
+    public static void main(String[] args) {
+        // You must either add the .NET developer certiifcate to the Java cacerts keystore, or uncomment the following
+        // lines to disable SSL validation.
+        //
+        // io.netty.handler.ssl.SslContext sslContext = io.netty.handler.ssl.SslContextBuilder
+        //     .forClient().trustManager(io.netty.handler.ssl.util.InsecureTrustManagerFactory.INSTANCE).build();
+        // httpClient = httpClient.secure(sslContextBuilder -> sslContextBuilder.sslContext(sslContext));
+
+        BlobClient blobClient = new BlobClientBuilder()
+            .connectionString(Configuration.getGlobalConfiguration().get("STORAGE_CONNECTION_STRING"))
+            .containerName("sample")
+            .blobName("sample.txt")
+            .retryOptions(new RequestRetryOptions(RetryPolicyType.FIXED, 3, Duration.ofMinutes(1),
+                Duration.ofSeconds(1), Duration.ofSeconds(1), null))
+            .addPolicy(new FaultInjectorUrlRewriterPolicy())
```

Comment:
Does this work in Java as a policy?  Including during retries?  In .NET, I believe we concluded we needed to use a transport instead of a policy, to guarantee it was the last thing run before sending the request to the HttpClient.

Reply:
A policy might work in .NET as well.  I will test it and discuss with the .NET team, and update the .NET storage sample if it seems to work.

Reply:
Either should work but an `HttpClient` would have better success. I'll add a sample for both